util/shlib_wrap.sh.in

```diff
@@ -117,6 +117,9 @@ if [ -f "$LIBCRYPTOSO" -a -z "$preload_var" ]; then
 fi
 {- output_on() unless grep (/-rpath\b/, @{$config{LDFLAGS}}); ""; -}
 
+OPENSSL_MODULES="${THERE}/providers"
+export OPENSSL_MODULES
+
```

Comment:
The trouble with this is that we may fool ourselves into relying on this, which is harmful on non-Unix platforms.

Reply:
I don't know about you, but I use this script myself all the time. I think it should just do "the right thing" without having to setup other env variables manually.

Reply:
I usually do things like `OPENSSL_MODULES=provider. /util/shlib_wrap.sh apps/openssl ...`

And look, the thing that made med think of this is the assignment of `OPENSSL_MODULES` in a test recipe, which is completely unnecessary in a Unix environment with the change you're making here. Come the day when we've become comfortable with this, and get annoyingly reminded by appveyor that oops, forgot.

Furthermore, there will be times when we may want to use this *without* getting `OPENSSL_MODULES` defined, to check some behavior bug for that scenario.

Reply:
> I usually do things like OPENSSL_MODULES=provider ./util/shlib_wrap.sh apps/openssl ...

That's going to become a big pain very quickly. Unlike with engines which we use relatively infrequently I think we're going to be doing this *all* the time.

> And look, the thing that made med think of this is the assignment of OPENSSL_MODULES in a test recipe, which is completely unnecessary in a Unix environment with the change you're making here. Come the day when we've become comfortable with this, and get annoyingly reminded by appveyor that oops, forgot.

Perhaps the issue here is that the script is being used for 2 different things. One as a wrapper for the tests, and one for developers from the command line. Perhaps we need a new script.

> Furthermore, there will be times when we may want to use this without getting OPENSSL_MODULES defined, to check some behavior bug for that scenario.

Well that one is easy. Given that's going to be very infrequent compared to the default case where we do want OPENSSL_MODULES defined - just don't use the script in those rare cases!

Anyway this whole issue is somewhat tangential to this PR. I think we need this for command line usage - but it isn't necessary for this particular PR. Since it is controversial I have removed it for now (I might bring it back in some future PR).

Reply:
> Well that one is easy. Given that's going to be very infrequent compared to the default case where we do want OPENSSL_MODULES defined - just don't use the script in those rare cases!

... 'cause any of the newly built programs are doing so well without the newly built libraries. 

> Perhaps we need a new script.

Yes, that's the better idea! 

> Since it is controversial I have removed it for now

Okie, cool